docs/user-guide/app_dependencies.md

```diff
@@ -0,0 +1,133 @@
+# Application dependencies
+
+Argo CD lets you natively model dependencies between Applications. This mechanism is intended to replace the prior way of modeling dependencies when using the app-of-apps pattern with sync waves on the Application custom resources.
+
+Using the new dependency mechanism, users are not bound to the app-of-apps pattern anymore and can define dependencies between applications using application selectors to define a directed, acyclic graph for dependency modeling. This means that an Application which defines one or more dependencies do not only depend on its direct dependencies, but to the dependencies of its dependencies as well.
+
+Application dependencies support both, manual and automatic sync policies.
+
+## Sync behaviour
+
+Defining dependencies for an Application will affect its sync behaviour. When a sync is started, Argo CD will evaluate the sync and health status of all other Applications in its dependency tree. Only if all of its dependent Applications have a sync status of `Synced` and a health status of `Healthy`, will the sync proceed. Until all conditions are met, the sync will stay in a progressing state and a message will indicate that the sync is waiting for dependencies. This behaviour is similar to sync waves, where a sync progresses only to the next wave when the current wave's resources are all synced and healthy.
+
+As a fallback mechanism, when using manual sync with the force option, Argo CD will not perform dependency resolution and the sync will proceed regardless of the states of the Application's dependencies.
+
+## Feature state and caveats
+
+This feature is in [beta state](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md#beta). It will be polished across the upcoming releases.
```

Comment:
I think this feature should start in alpha behind a feature flag. If the feature already included a way to handle cross-project dependencies, provided a full UI/CLI experience, and handled things like warning on deletion of a dependency app, I think it would make sense to start in beta.